### DISCUSSION TITLE
Error Row connectivity

### DISCUSSION THREAD

**Discussion by arnalpascal** [Yowkees/keyball#433] [Q&A]
Hi,

I soldered the 2 plates, diodes, sockets and controlers and do a test. All rows of the 2 plates seems to be ok except one (with the tape).
I tested connectivity from each socket to pro micro and it seem to be ok
Is-it possible that the pro micro is faulty ?
![PXL_20231201_163533689](https://github.com/Yowkees/keyball/assets/50094748/926c956c-8213-498d-8b00-ac18e879f9c7)
![PXL_20231201_163600696](https://github.com/Yowkees/keyball/assets/50094748/78e3b48e-678c-41bc-90e8-34a05fa40e99)
![PXL_20231201_163645642](https://github.com/Yowkees/keyball/assets/50094748/5f55c799-422c-4d7a-ad67-99e611a2805a)

Thanks,

**Comment by Yowkees** [comment]
Hi. Thank you for choosing Keyball61:-)

At first, please check soldering. In this case, "ROW3" line has problem.
I think Red circled soldering in below picture is too big. Please reheat here.

If the problem is not resolved, please switch right and left Promicro to another side. It will help to know ProMicro's faulty.

Thank you very much.
![KB61_ProMicroFix](https://github.com/Yowkees/keyball/assets/84209772/1991f093-46cc-4e0f-ad8d-774a38e6f51b)

**Comment by arnalpascal** [comment]
Hi,

Thanks for your reply.
I try to unsolder the controler (no success for now) but remove a lot of soldering tin.

However, I have a new information.
All keys except this row continue to be ok, but for this row, the 3 rightest keys display all characters of their respective colomn when I test it with my continuity tester. The 4 leftest continue to not respond, and the tester bip.
But when I test with a piece of soldering tin, no letter is display on my computer for the entire row.

Do you have any suggestion before a try to remove the controler please ?

![PXL_20231206_081544685](https://github.com/Yowkees/keyball/assets/50094748/b5057146-afed-44ee-ab57-ecbda8c1e112)


Thanks

**Comment by Yowkees** [comment]
Hi.

Could you check all green points are connected in below picture?
And one more curious, is there any solder residue in red circle point in below picture.
This problem comes from "ROW3" line. Red circled through hole is ROW3. please check ROW3 is not connected next through hole.

Thank you.
![kb61_row3fix](https://github.com/Yowkees/keyball/assets/84209772/632d12a7-2ba9-46cd-9aa6-092a0baad357)

**Comment by arnalpascal** [comment]
Hi,

Thanks a lot, it was the red circle the problem.
Now, all key seems to be OK.

Next step, DEL soldering.

 Do you think my ugly soldering below the red circle will be a problem for the trackball please ?

Thanks again,

Can't wait to test it when it will be finish.

**Comment by arnalpascal** [comment]
Hi,
New big big problem.
I think i done a big mistake.
I tried to unsolder the first led of the left plate because of an ugly soldering and since, nothing is ok.
I can't reflash this keyboard part, nothing happen.

![PXL_20231217_081544685](https://github.com/Yowkees/keyball/assets/50094748/bc3cbfd0-5820-4bb6-a1c7-f710143fda04)

Is there a solution please ?

Thanks

**Comment by Yowkees** [comment]
Hi.

I think some of the LED terminals are shorted.
That's why ProMicro doesn't start.

At this morment, please Ignore LED1 and please check other LED terminals that are shorted to their neighbors and remove them.

Thanks.

**Comment by arnalpascal** [comment]
Hi,

I tried to remove some leds carrefully, but no success, nothing happen.

![IMG_9354](https://github.com/Yowkees/keyball/assets/50094748/6666710d-5683-4187-8cd6-99900646ff0a)

Any idea please ?

Thanks

**Comment by Yowkees** [comment]
Hi. I understood your situation.

I think it's time to cut off LED power lines from PCB temporarily.
Maybe, some LEDs are broken as short mode by heat.
Please see attached pictures.

Please scratch (grind) off red lined PCB pattarn by using back of box cutter (Yellow circled part on second picture).
Please scratch along the red line again and again.

If right and left square pcb patterns are not shorted, it's OK.
All LEDs are separated from ProMicro. ProMicro will work again.

If you want to try LED soldrig again, just put solder here as other jumper soldering.

![kb61_ledcut](https://github.com/Yowkees/keyball/assets/84209772/c83e83fb-c2a4-47d8-bd16-f127bbc4548e)
![cutter](https://github.com/Yowkees/keyball/assets/84209772/97ec41b4-26e0-4688-8895-65097a5bd704)


Good luck!

**Comment by arnalpascal** [comment]
Hi,

Thanks a lot, I cut off the led as mentionned, and the left plate restart.

So now, the 2 plates are OK individualy.

However, I'm not be able to have the complete keyboard  OK, just the side that I plug to the computer is OK.
![PXL_20240106_142408972 MP](https://github.com/Yowkees/keyball/assets/50094748/f5a56a0e-7ef4-46e7-b5cb-c590ba249a97)


Could you please tell me what is wrong ?

Thanks a lot again.

**Comment by arnalpascal** [comment]
Hi,
I resoldered the data edges of the 2 promicro and the data passes from one part to the other.
However, if I plug the right plate with the trackball with USB , the order is OK, but if I plug the USB on the left, then the 2 are reversed (trewq on second row of the rigth plate).
I thinks it's a firmware / eeprom problem because I try several things from qmk before I resoldered data?

Best Regards

**Comment by Yowkees** [comment]
Hi,
If there is a problem on DATA line, please re-check the soldering of TRS connector. And please check LEFT jumper too.
Thanks.
